execution/block-partitioner/src/sharded_block_partitioner/conflict_detector.rs

```diff
@@ -45,9 +45,24 @@ impl CrossShardConflictDetector {
         let mut accepted_txns = Vec::new();
         let mut accepted_txn_dependencies = Vec::new();
         let mut rejected_txns = Vec::new();
+        let mut discarded_senders = HashSet::new();
         for (_, txn) in txns.into_iter().enumerate() {
             if self.check_for_cross_shard_conflict(self.shard_id, &txn, cross_shard_rw_set) {
+                if let Some(sender) = txn.sender() {
+                    discarded_senders.insert(sender);
+                }
                 rejected_txns.push(txn);
+            } else if let Some(sender) = txn.sender() {
+                if discarded_senders.contains(&sender) {
+                    rejected_txns.push(txn);
+                } else {
+                    accepted_txn_dependencies.push(self.get_deps_for_frozen_txn(
+                        &txn,
+                        Arc::new(vec![]),
+                        prev_rounds_rw_set_with_index.clone(),
+                    ));
+                    accepted_txns.push(txn);
+                }
```

Comment:
nit: maybe save a few lines like this:

```suggestion
            let sender_was_discarded = txn.sender().map_or(|sender| discarded_sender.contains(&sender), false);
            if sender_was_discarded || self.check_for_cross_shard_conflict(self.shard_id, &txn, cross_shard_rw_set) {
                if let Some(sender) = txn.sender() {
                    discarded_senders.insert(sender);
                }
                rejected_txns.push(txn);
```